src/packet/mod.rs

```diff
@@ -78,9 +84,11 @@ impl PrimitiveValues for ::std::net::Ipv6Addr {
     }
 }
 
+pub mod checksum;
 pub mod ethernet;
 pub mod ip;
 pub mod ipv4;
 pub mod ipv6;
 pub mod udp;
```

Comment:
The checksums in UDP will need to be converted to this new method of checksumming too.